### PR TITLE
Adds initial code for adding and removing a package reference

### PR DESCRIPTION
**Pull request by Shazwazza**
Hi,

At the MVP Summit Hackathon I created this ability via a dotnet extension however I realize that this functionality is supposed to be part of the core cli code so I'm submitting this as a PR.

Related Issue: https://github.com/dotnet/cli/issues/4521 

I know this doesn't complete the functionality required for 4521 but it's a start.

##Some notes about this:

* This only deals with package references, not project references. 
* This follows the syntax as specified in https://github.com/dotnet/cli/issues/4521 
* It will add or update a project reference with using `dotnet ref add`
* Originally I did this using an MSBuild extension which allowed me to automatically get the location of the csproj, but to do this in the CLI project meant adding a few references in places they didn't already exist so I opted to finding the csproj file manually - I tried following the same format that's currently in this codebase so hopefully that is satisfactory
* I could not get the entire project to build with the build.cmd script. It mostly builds but fails eventually with

```
\TestAssets\TestPackages\dotnet-desktop-binding-redirects\project.json(7,30): error NU1001: The dependency Newtonsoft.Json >= 5.0.0 could not be resolved. [X:\Projects\NETCLI\Source\build.proj]
```

* I wanted to create unit tests for this, however I could not get the other unit tests to run correctly. I tried to build some test projects manually such as: `/test/dotnet-new.Tests` by running `dotnet restore`, `dotnet build`, `dotnet test` but running the tests fail so I figured I must be missing something in order to get these tests to run and to use for my own.

##Usage

- this will default to using "*" as the version
`dotnet ref add NewtonSoft.Json`
- to specify specific version
`dotnet ref add NewtonSoft.Json 9.0.1`
- to remove
`dotnet ref del NewtonSoft.Json`

### CODE REVIEW

**Comment by piotrpMSFT** [comment]
@krwq this can be helpful to the work you and @blackdwarf were discussing. I know the invocation pattern is likely to change but the underpinnings are here.

**Comment by krwq** [comment]
@dotnet-bot test  Fedora23 x64 Debug Build please

**Comment by krwq** [comment]
@piotrpMSFT - can we merge this then and then do the changes or just cherry-pick and pick up from here?

**Comment by krwq** [comment]
Thanks @Shazwazza for the contribution! I will pick up from here since we did some design changes (name of the command etc) and closing this PR for now - expect a new PR from me with your commit

File: src/dotnet/commands/dotnet-ref/Program.cs [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+    public class RefCommand
+    {
+        public static int Run(string[] args)
+        {
+            DebugHelper.HandleDebugSwitch(ref args);
+
+            DirectoryInfo currDir = new DirectoryInfo(Directory.GetCurrentDirectory());
+            FileInfo projectFile = currDir.GetFiles("*.csproj").FirstOrDefault();
+
+            if (!projectFile.Exists)
```
Comment (eerhardt) [inline src/dotnet/commands/dotnet-ref/Program.cs]: This should happen later. If this check happens here, a user can't execute the "help" command to figure out what is wrong.

File: src/dotnet/commands/dotnet-ref/Program.cs [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+
+namespace Microsoft.DotNet.Tools.Ref
+{
+    public class RefCommand
+    {
+        public static int Run(string[] args)
+        {
+            DebugHelper.HandleDebugSwitch(ref args);
+
+            DirectoryInfo currDir = new DirectoryInfo(Directory.GetCurrentDirectory());
```
Comment (eerhardt) [inline src/dotnet/commands/dotnet-ref/Program.cs]: The project should be an optional parameter. And check the current directory if the project isn't specified.

**Comment by Shazwazza** [comment]
👍 great! I'm not sure if there's some additional information I've missed regarding being able to get the build.cmd working but if you have any pointers I'd love to know why it's failing on my machine (maybe it's just in-flux atm)

**Comment by piotrpMSFT** [comment]
@Shazwazza can you open an issue with the build failures you're seeing? Builds are always green before we merge...

**Pull request closed without merging**